packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -582,6 +614,8 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
         defaultProps: {
           className: Dropdown.slotClassNames.item,
           active: highlightedIndex === index,
+          selected: !this.props.multiple && value === item,
```

Comment:
this is based on potentially dangerous assumption that here

```js
const filteredItems = this.getItemsFilteredBySearchQuery()
```

identity of each `item` in `this.props.items` won't be changed. However, given that client's code is providing this logic, it is possible that this won't be true, and chances are higher in case if any library based on immutability principles is used (such as Redux). 

Simplest example is:

```jsx
<Dropdown search={items => [...items]} />
```

Reply:
I had this prev implemented with using state's selectedIndex to track this. Do you think this will be better option?

Reply:
I think that better one would be to provide some unique key to each item (if those are objects), like `_someKey` prop, the one that will be used for identity check. Something like the following (pseudo):

```jsx
const getItemsFilteredBySearchQuery = () => { 
  const itemsToFilter = augmentWithSomeKey(this.props.items)
  ...
  const filteredItems = search(itemsToFilter)
  return filteredItems
}
```

and then, before rendering, strip this helper `_someKey` prop:

```jsx
const renderItems = () => {
  const filteredItemsWithNoAugmentedProps = stripAugmentedProps(filteredItems)
  .. render filteredItemsWithNoAugmentedProps ..
}
```

This would be the safest option.